### ISSUE TITLE
Persist changes to menu? After additems or removeItems? How?

### ISSUE THREAD

**Issue by WriterStat** [adgsm/multi-level-push-menu#111]
How can one persist changes to the menu?

Additems and removeitems doesn't seem to make much sense if we can't persist the changes to the menu.

Can one get the current menu array?

On a change of the menu?  On either an additems or removeitems?

For storage.  And reuse?

How to get the resulting menu array or menu changes for reuse later?

Possible?  Is there a demo for persisting changes?

Thanks,

**Comment by adgsm** [comment]
You can easily add such functionality into createItem method... What stops you to simply push additions to the array you've used to create the menu?

**Comment by WriterStat** [comment]
I had these same thoughts myself last night., just wanted to make sure I was polite and asked first.  

Probably my lack of javascript knowledge.  New to it, but learning fast.  Multi-dimensional object arrays in javascript kill me. =) Wrapping my head around it. 

I've written this:

$( '#add_menu_item' ).click(function(){
		item_counts++;
		var item_position = item_counts -1;
		var item_name = 'Item '+item_counts;		
		var add_to = $( '#menu' ).multilevelpushmenu( 'findmenusbytitle' , 'MyItems' ).first();
		var add_item = [{name: item_name, id: item_name, link: '#'}];
		$( '#menu' ).multilevelpushmenu( 'additems' , add_item , add_to , item_position );
	});

Where $( '#add_menu_item' ) is actually in your menu to add items.

I think I can add the slice/splice to arrMenu to the code above and the save to local storage.  I've already written the LocalStorage code.  Thus encapsulating your menu code, so I don't have to change your fine code.

I think I'm just stopped, because of my lack of understanding Multi-dimensional object arrays and how to get stuff back in.

You my friend are a javascript super hero. =)

Basically how can I slice or splice the change, in the right place, either the add or remove back into arrMenu?

I understand Array.push(), but not yet the logic to put it or remove the item in the right place in arrMenu.

I think once I understand this then I can fork the multi-menu code and actually put it in the code for you. Once I get it working outside.  Understand it.

**Comment by WriterStat** [comment]
I am thinking of using this...

pos = arrMenu.map(function(e) { return e.id; }).indexOf('item10');

or this...

function arrayObjectIndexOf(myArray, searchTerm, property) {
    for(var i = 0, len = myArray.length; i < len; i++) {
        if (myArray[i][property] === searchTerm) return i;
    }
    return -1;
}
pos = arrayObjectIndexOf(arrMenu, "item10", "id");

To get the index of an item if needed.  And I've decided to keep an outside external array of objects, that contains meta data about the Menu.  Things like item counts etc...  for menu areas, that can be added to.

I've decided to add ids to all menu items.  In case the names change.  That way I can interrogate or find things in the actual menu and maybe even allow people to drag and drop menu items to change order etc...

There's probably a better way, but this is all I could think of.

Let me know if I'm way off.

**Comment by WriterStat** [comment]
I had the same thought, last night.  But wanted to see if there was a better way, within the current code base to get the current menu array.

I've created the code methods to update the arrMenu that's originally passed in to create the menu, with real time changes.  Turns out when adding many/multiple nested menu items/menu areas to the Menu, it's much more efficient to add them all at once to the arrMenu object array and then reinitialize the menu display, based on the new changed arrMenu. Than it is to add them one at a time using the current code Menu's addItem method.  Now working on a way to reinitialize the menu display based on arrMenu after changes. Just seems more efficient.

If there's ever an update/rewrite to this, you should keep an updated underlying array for the menu.  And maybe separate the display from data.
